mne/encoding/model.py

```diff
@@ -0,0 +1,238 @@
+"""Base and high-level classes for fitting encoding models."""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#          Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+from ..externals.six import string_types
+
+
+class SampleMasker(object):
+    """Remove samples with a particular value before fitting.
+
+    Parameters
+    ----------
+    estimator : instance of sklearn-style estimator
+        The estimator to be called after samples are removed.
+    mask_val : float, int, nan | callable | None
```

Comment:
I could call it `mask_values` and then let the user supply a list of objects that will be used for removal. (e.g., you could do `mask_values=[np.nan, 0.]`